api/api.py

```diff
@@ -270,7 +271,13 @@ def chunked_completion(self, completion_request: CompletionRequest):
         )
         generator_args = GeneratorArgs(
             completion_request.messages[-1].get("content"),
+            max_new_tokens=(
+                int(completion_request.max_tokens)
```

Comment:
why cast?

Reply:
The request object was being populated from JSON, which automatically means the value for every field is str despite the dataclass type hints. 

These casts were to get things working, but yes - this is a bad solution and we should have better type enforcement here (and possibly repo-wide)

Based on what I can find, we can resolve this by casting the types at the object's init time for each field (kinda tedious) or by using something like pydantic (yet another package to manage). 

Should we attack this in another issue/PR or try to resolve it here?

Reply:
Casting at init time makes sense, since downstream users shouldn't need to think about it

We can make it a separate PR or push the casting to init just for max_tokens and temperature

Reply:
Created https://github.com/pytorch/torchchat/issues/1025 to track this, will merge this PR for now.